eiffel-vocabulary/EiffelIssueDefinedEvent.md

```diff
@@ -0,0 +1,278 @@
+<!---
+   Copyright 2018 Jaden Young
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+--->
+
+# EiffelIssueDefinedEvent (ID)
+The EiffelIssueDefinedEvent declares that an issue has been created in some
+external issue management software. This event provides provenance for an issue,
+and acts as a "handle" for future issue related events similarly to
+[EiffelActivityTriggeredEvent](./EiffelActivityTriggeredEvent.md) and
+[EiffelTestCaseTriggeredEvent](./EiffelTestCaseTriggeredEvent.md). It provides
+a minimal amount of information necessary for this purpose and allows further
+data to be attached through later events, for example the 
+[EiffelIssueAssignedEvent](./EiffelIssueAssignedEvent).
+
+## Data Members
+
+### data.type
+__Type:__ String  
+__Required:__ Yes  
+__Legal values:__ BUG, IMPROVEMENT, FEATURE, WORK_ITEM, REQUIREMENT, OTHER  
```

Comment:
Should this really be mandatory? And what if the type is changed in the issue management system, how should that be handled? I'd propose 'initialType' or 'initialCategory' and make it free text. It's values can be enforced by implementation specific guidelines instead.

Reply:
Eh, I just saw that we have called this 'type' in the current issue references and the legal values come from there. It's not your invention. Sorry :) Anyway, when we're now at it I still think this should not be mandatory and its values should be free text.

Reply:
The type changing in the issue management system is a good point. Similar to ISM, it's an instance of the question "Should Eiffel care about modifications to an entity's data?" which in turn depends on "To what degree does Eiffel care about describing an entity, and when does Eiffel simply identify it?" The latter question is answered in other events, for example ED and SCC, but has yet to be decided for issue related events.

This also relates to the earlier discussion about the semantic similarities of ID to FCD/ED vs ActT/TCT. The closer towards the identification side ID goes, the more similar to FCD, the closer to the descriptive side the closer to ActT.

As far as the values of `type` being an enum or free form string, when you say "implementation specific guidelines," are you talking about an implementation of the Eiffel protocol, i.e. "Within our company these are the legal values," or implementation of the issue management system, i.e. `data.tracker`?

Reply:
I agree that an Eiffel deployment needs to have a common vocabulary. I'm just not sure all of it needs to be defined in the protocol itself. In some parts it can be more convenient to define the legal values in guidelines/wrappers to be used within the company/deployment. I completely agree that we don't want different values on data.type from different issue trackers within the same company/deployment.

Reply:
Couldn't that be said about basically anything, though? I'd like some unambiguous rules as to what gets to be defined in the protocol, and what is left to guidelines/wrappers in each deployment. Right now it's sort of a gradient where we end up at slightly different places along the axis with every judgment call we make.

Reply:
Agreed. And that's a topic to discuss in another issue.
About this one, let's keep the type values there for now, since they are already part of the current events (IV and SCC). And that field is mandatory in those events too, so let's keep it mandatory for now.